python/lsst/meas/astrom/pessimistic_pattern_matcher_b_3D.py

```diff
@@ -0,0 +1,1111 @@
+              max_n_patterns, max_shift, max_rotation, max_dist,
+              min_matches, pattern_skip_array=None):
+        r"""Match a given source catalog into the loaded reference catalog.
+
+        Given array of points on the unit sphere and tolerances, we
+        attempt to match a pinwheel like pattern between these input sources
+        and the reference objects this class was created with. This pattern
+        informs of the shift and rotation needed to align the input source
+        objects into the frame of the refernces.
+
+        Parameters
+        ----------
+        source_array: float array
+            An array of spherical x,y,z coordinates and a magnitude in units of
+            objects having a lower value for sorting. The array should be of
+            shape (N, 4).
+        n_check : int value
+            Number of sources to create a pattern from. Not all objects may be
+            checked if n_match criteria is before looping through all n_check
+            objects.
+        n_match : int value
+            Number of objects to use in constructing a pattern to match.
+        n_agree: int value
+            Number of found patterns that must agree on their shift and
+            rotation before exiting. Set this value to 1 to recover the
+            expected behavior of Optimistic Pattern Matcher B.
+        max_n_patters : int value
+            Number of patterns to create from the input source objects to
+            attempt to match into the reference objects.
+        max_shift: float value
+            Maximum allowed shift to match patterns in arcseconds.
+        max_rotation: float value
+            Maximum allowed rotation between patterns in degrees.
+        max_dist: float value
+            Maximum distance in arcseconds allowed between candidate spokes in
+            the source and reference objects. Also sets that maximum distance
+            in the intermediate verify, pattern shift/rotation agreement, and
+            final verify steps.
+        pattern_skip_array: int array
+            Patterns we would like to skip. This could be due to the pattern
+            being matched on a pervious iteration that we now consider invalid.
+            This assumes the ordering of the source objects is the same between
+            different runs of the matcher which, assuming no object has been
+            inserted or the magnitudes have changed, it should be.
+
+        Returns
+        -------
+        output_struct : pipe.base.struct
+            A pipebase struct containing the following outputs
+                maches : a (N, 2) int array of matched pairs
+                distances : a float array of pair distances
+                pattern_idx : int index of matched pattern
+                shift : a float value for the shift between the source and
+                    reference objects.
+        Returns None type if no match is found.
```

Comment:
Actually, never mind. I have thought of a way this is useful. I will make the change to only return a struct.